src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -96,5 +86,236 @@ public bool NeedsRefresh()
         {
             return false;
         }
+
+        private FormatException CreateFeatureFlagFormatException(string jsonPropertyName, string settingKey, string foundJsonValueKind, string expectedJsonValueKind)
+        {
+            return new FormatException(string.Format(
+                ErrorMessages.FeatureFlagInvalidJsonProperty,
+                jsonPropertyName,
+                settingKey,
+                foundJsonValueKind,
+                expectedJsonValueKind));
+        }
+
+        private FeatureFlag ParseFeatureFlag(string settingKey, string settingValue)
+        {
+            FeatureFlag featureFlag = new FeatureFlag();
+
+            var reader = new Utf8JsonReader(System.Text.Encoding.UTF8.GetBytes(settingValue));
+
+            try
+            {
+                reader.Read();
+
+                if (reader.TokenType != JsonTokenType.StartObject)
+                {
+                    throw new FormatException(string.Format(ErrorMessages.FeatureFlagInvalidFormat, settingKey));
+                }
+
+                while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+                {
+                    if (reader.TokenType != JsonTokenType.PropertyName)
+                    {
+                        continue;
+                    }
+
+                    string propertyName = reader.GetString();
+
+                    switch (propertyName)
+                    {
+                        case FeatureManagementConstants.IdJsonPropertyName:
+                            {
+                                if (reader.Read() && reader.TokenType == JsonTokenType.String)
+                                {
+                                    featureFlag.Id = reader.GetString();
+                                }
+                                else if (reader.TokenType == JsonTokenType.Null)
+                                {
+                                    // Effectively skips the feature flag if the Id is null
+                                    return null;
```

Comment:
This is putting business logic in the deserialization code path. I'd rather see the flag returned and the caller check if id is null or not.

Reply:
Makes more sense that way, will move it out